vlad.metelap/task1.rb

```diff
@@ -0,0 +1,30 @@
+class Task1
+
+  b = Array.new(3, Array.new(3, 0))
+  # b = [[0,0,0],[0,0,0],[0,0,0]]
+  b = b.map {|el| el.map {rand(-10..10)}}
+  positivearr = b.map {|el| el.map {|elm| elm > 0 ? elm : 0} }
```

Comment:
You can use method `positive?` instead of condition `elm > 0`

https://ruby-doc.org/core-2.5.0/Numeric.html#method-i-positive-3F

Reply:
Also, to prevent iteration of nested arrays, I recommend you to use method `flatten` 

https://apidock.com/ruby/Array/flatten

Reply:
done